cmd/lotus-shed/dagspliter/dagsplitter.go

```diff
@@ -325,13 +390,63 @@ var Cmd = &cli.Command{
 			fmt.Printf("%s\n", boxCids[i])
 		}
 
+		CAR_OUT_DIR := "dagsplitter-car-files"
+		if _, err := os.Stat(CAR_OUT_DIR); os.IsNotExist(err) {
+			os.Mkdir(CAR_OUT_DIR, os.ModePerm)
+		}
+
+		for _, boxCid := range boxCids {
+			box := Box{}
+			if err := cst.Get(ctx, boxCid, &box); err != nil {
+				// FIXME: We could just retain the created boxes but trying
+				//  to make it more decoupled (we might not have them available
+				//  by the time this is called).
+				return xerrors.Errorf("failed retrieving box: %w", err)
+			}
+
+			//_, _ = fmt.Fprintf(os.Stderr, "Creating car with roots: %v\n", box.Nodes)
+
+			out := new(bytes.Buffer)
+			if err := car.WriteCarWithWalker(context.TODO(), dag, box.Nodes, out, BoxCarWalkFunc(&box)); err != nil {
+				return xerrors.Errorf("write car failed: %w", err)
+			}
+
+			if err := ioutil.WriteFile(filepath.Join(CAR_OUT_DIR, boxCid.String()+".car"),
+				out.Bytes(), 0644); err != nil {
+				return xerrors.Errorf("write file failed: %w", err)
+			}
+		}
+
 		return nil
 	},
 }
 
+func BoxCarWalkFunc(box *Box) func(nd ipld.Node) (out []*ipld.Link, err error) {
+	return func(nd ipld.Node) (out []*ipld.Link, err error) {
+		for _, link := range nd.Links() {
+
+			// Do not walk into nodes external to the current `box`.
+			if box.isExternal(link) {
+				//_, _ = fmt.Fprintf(os.Stderr, "Found external link, skipping from CAR generation: %s\n", link.Cid.String())
+				continue
+			}
+
+			// FIXME: Taken from the original `gen.CarWalkFunc`, what are these?
```

Comment:
Filecoin sector commitment CIDs (CommD (padded/truncated sha256 binary tree), CommR (basically magic tree))

Those are linked directly in the chain state, so this avoids trying to accidentally walk over a few exabytes of data

Reply:
Ok, so we leave this check. (Ideally this walk function should reuse the code from the original one to avoid duplicating these).